### DISCUSSION TITLE
[Hardware] SoftwareBitBang, distance?

### DISCUSSION THREAD

**Discussion by scottyob** [gioblu/PJON#417] [General]
For a 5V ATTiny85 setup at each node, I'm reading the crazy distances and am wondering realistically what I'd need to do to get more than a few hundred feet? (18 gauge wire).  I assume the voltage drop would be crazy?  What hardware do you use for long distances?  Do I need to switch between higher voltages?

**Comment by gioblu** [comment]
Using PJDL, with a direct connection between io pins, and 5v voltage level, you can reach up to 2000m.
GIve it a try :)

**Comment by fredilarsen** [comment]
In my experience, the long distances more easily pick up induced voltage from lightning etc, so I would recommend both a series resistor (180 Ohm) and a pulldown resistor on each device.

I had one of my longer outdoor wires pick up voltage enough to destroy several of my attached Arduino Nanos after a nearby lightning that hit nothing. So I learned the hard way to add the resistors. I also added a 50mA fast glass fuse (5x20mm) in series, plus a 6V Zener (I think i changed that to a TVS diode) between signal and ground, on the device side of the fuse. Probably the resistors are enough, the series resistors also protecting against bus short circuit. But I was a little paranoid and added the fuse and TVS as well. System has been running fine through many lightning storms for several years after.

A comment to the voltage drop question: the voltage drop over the wire will increase with increasing current, but the only current is from the signal pin and pulldown, and this is just microamps.

**Comment by thilojaeggi** [comment]
Does SoftwareBitBang reach a longer distance than ThroughSerial though?
I assume ThroughSerial has higher throughput though, also are there any other differences, they both seem to have error correction etc.
